### ISSUE TITLE
Changing a spaces quota does not get applied

### ISSUE THREAD

**Issue by gordcorp** [cloudfoundry-community/terraform-provider-cf#184]
I've noticed that updating which space quota is assigned to a space  does not properly get applied.

An example:
```
resource "cloudfoundry_org" "test_org" {
  name = "test-org"
}

resource "cloudfoundry_space" "test_space" {
  name  = "test-space"
  org   = "${cloudfoundry_org.test_org.id}"
  quota = "${cloudfoundry_space_quota.10g.id}"
}

resource "cloudfoundry_space_quota" "10g" {
  name                     = "10g"
  allow_paid_service_plans = false
  total_memory             = 10240
  total_routes             = 5
  total_services           = 20
  org                      = "${cloudfoundry_org.test_org.id}"
}

resource "cloudfoundry_space_quota" "20g" {
  name                     = "20g"
  allow_paid_service_plans = false
  total_memory             = 20240
  total_routes             = 5
  total_services           = 20
  org                      = "${cloudfoundry_org.test_org.id}"
}
```

I can see that the "10g" is properly applied.

```
$ cf space test-space | grep quota
space quota:               10g
```

If we update the space to the other quota:
```
resource "cloudfoundry_space" "test_space" {
  name  = "test-space"
  org   = "${cloudfoundry_org.test_org.id}"
  quota = "${cloudfoundry_space_quota.20g.id}"
}
```

And then terraform plan and apply that, it still shows the old space quota applied with ` cf space test-space | grep quota`. Running terraform plan again shows the change is still pending. To make the plan go clean as a workaround I can set the quota manually with.
```
$ cf unset-space-quota test-space 10g
Unassigning space quota 10g from space test-space as admin...
OK
$ cf set-space-quota test-space 20g
Assigning space quota 20g to space test-space as admin...
OK
```

This is against the latest cf-deployment v6.10.0, terraform 0.11.11.

**Comment by gberche-orange** [comment]
Thanks @gordcorp for reporting this issue, and sorry for late response on this. 

Indeed it seems that the provider is properly using [POST /v2/spaces/](http://apidocs.cloudfoundry.org/7.1.0/spaces/creating_a_space.html) to create a space with an initial space quota definition guid associated, but that during a space update, as the [PUT /v2/spaces/:guid](http://apidocs.cloudfoundry.org/7.1.0/spaces/update_a_space.html) is lacking the ability to update the associated space quota definition guid, it currently lacks the invocation to the [PUT /v2/space_quota_definitions/:guid/spaces/:space_guid](http://apidocs.cloudfoundry.org/7.1.0/space_quota_definitions/associate_space_with_the_space_quota_definition.html) endpoint (that the `cf set-space-quota` command invokes)

https://github.com/mevansam/terraform-provider-cf/blob/2fc7c677c116c864fad331b589921d7b5b630b73/cloudfoundry/cfapi/space_manager.go#L166-L192
https://github.com/mevansam/terraform-provider-cf/blob/2fc7c677c116c864fad331b589921d7b5b630b73/cloudfoundry/resource_cf_space.go#L201-L223

The space update use-case it not yet covered by the associated unit test at 
https://github.com/mevansam/terraform-provider-cf/blob/dev/cloudfoundry/resource_cf_space_test.go

**Comment by ArthurHlt** [comment]
This has been fixed in #203, i'm closing it has it would require more works on old provider
